gammapy/maps/wcs.py

```diff
@@ -551,6 +556,47 @@ def to_cube(self, axes):
         axes = copy.deepcopy(self.axes) + axes
         return self.__class__(self._wcs.deepcopy(), npix, cdelt=cdelt, axes=axes)
 
+    def pad(self, pad_width):
+        if np.isscalar(pad_width):
+            pad_width = (pad_width, pad_width)
+        npix = (self.npix[0] + 2 * pad_width[0],
+                self.npix[1] + 2 * pad_width[1])
+        wcs = self._wcs.deepcopy()
+        wcs.wcs.crpix += np.array(pad_width)
+        return self.__class__(wcs, npix, cdelt=copy.deepcopy(self._cdelt),
+                              axes=copy.deepcopy(self.axes))
+
+    def crop(self, crop_width):
+        if np.isscalar(crop_width):
+            crop_width = (crop_width, crop_width)
+        npix = (self.npix[0] - 2 * crop_width[0],
+                self.npix[1] - 2 * crop_width[1])
+        wcs = self._wcs.deepcopy()
+        wcs.wcs.crpix -= np.array(crop_width)
+        return self.__class__(wcs, npix, cdelt=copy.deepcopy(self._cdelt),
+                              axes=copy.deepcopy(self.axes))
+
+    def downsample(self, factor):
+
+        if (not np.all(np.mod(self.npix[0], factor) == 0) or
+                not np.all(np.mod(self.npix[1], factor) == 0)):
+            raise ValueError('Data shape is not divisible by {} in all axes.'
+                             ' Pad image prior to downsampling to correct'
+                             ' shape.'.format(factor))
+
+        npix = (self.npix[0] / factor, self.npix[1] / factor)
+        cdelt = (self._cdelt[0] * factor, self._cdelt[1] * factor)
+        wcs = get_resampled_wcs(self.wcs, factor, True)
+        return self.__class__(wcs, npix, cdelt=cdelt,
```

Comment:
I think it might be a good idea to add a `copy` method, and to always go through that copy method in other methods that make new, similar maps. The problem with the current pattern is that it's hard to maintain / be consistent when we add properties like unit or meta, it's easy to forget to update some of these `self.__class__` callers. 

A copy method that by default makes a copy of all data members in a single place is easier to maintain. So by default it copies everything, but if the caller passes in things they want differnt about the copy (like here `wcs`), then those are used to construct the new object.

I haven't fully thought this through, so I'm not sure it's a good suggestion that really helps. @woodmd - Thoughts?

Reply:
A `copy` method seems like it could be useful but I think requires some careful thought.  Is this an idiom that you use elsewhere in gammapy or have you seen other packages use it?  My feeling is that this would probably better fit in a separate PR where we can have a dedicated discussion on the topic.

Reply:
> Is this an idiom that you use elsewhere in gammapy or have you seen other packages use it?

Not yet. It's something I wanted to introduce after seeing issues with methods that construct new objects in other places, e.g. in SkyImage / SkyCube.

> My feeling is that this would probably better fit in a separate PR where we can have a dedicated discussion on the topic.

Agreed. You could leave a `# TODO` comment, although if you always use `self.__class__` in gammapy.maps it's also easy to find all cases via text search.